src/widgets/immunizations/immunizations-form.component.tsx

```diff
@@ -0,0 +1,342 @@
+import React, { useEffect, useRef, useState } from "react";
+import SummaryCard from "../../ui-components/cards/summary-card.component";
+import styles from "./immunizations-form.css";
+import { DataCaptureComponentProps } from "../shared-utils";
+import { useTranslation } from "react-i18next";
+import { savePatientImmunization } from "./immunizations.resource";
+import { mapToFHIRImmunizationResource } from "./immunization-mapper";
+import { useCurrentPatient } from "@openmrs/esm-api";
+import { useHistory } from "react-router-dom";
+import { createErrorHandler } from "@openmrs/esm-error-handling";
+import { getStartedVisit } from "../visit/visit-utils";
+import useSessionUser from "../../utils/use-session-user";
+import {
+  ImmunizationFormData,
+  ImmunizationSequence
+} from "./immunization-domain";
+
+export function ImmunizationsForm(props: ImmunizationsFormProps) {
+  const [vaccineName, setVaccineName] = useState("");
+  const [vaccineUuid, setVaccineUuid] = useState("");
+  const [immunizationObsUuid, setImmunizationObsUuid] = useState("");
+  const [vaccinationDate, setVaccinationDate] = useState(null);
+  const [immunizationSequences, setImmunizationSequences] = useState([]);
+  const [currentDose, setCurrentDose] = useState<ImmunizationSequence>(
+    {} as ImmunizationSequence
+  );
+  const [vaccinationExpiration, setVaccinationExpiration] = useState(null);
+  const [lotNumber, setLotNumber] = useState(undefined);
+  const [manufacturer, setManufacturer] = useState("");
+  const [enableCreateButtons, setEnableCreateButtons] = useState(false);
+  const [enableEditButtons, setEnableEditButtons] = useState(true);
+  const [viewEditForm, setViewEditForm] = useState(true);
+  const [formChanged, setFormChanged] = useState<Boolean>(false);
+  const [
+    isLoadingPatient,
+    patient,
+    patientUuid,
+    patientErr
+  ] = useCurrentPatient();
+  const formRef = useRef<HTMLFormElement>(null);
+  const { t } = useTranslation();
+  const history = useHistory();
+  const today = new Date().toISOString().split("T")[0];
+  const currentUser = useSessionUser();
+
+  useEffect(() => {
+    if (vaccinationDate) {
+      setEnableCreateButtons(true);
```

Comment:
If I understand correctly, you are saying I should get rid of this `useEffect` here. And probably use this code of `if(vaccinationDate){}` after the `setVaccinationDate` is called.
Is my understanding correct Or I missed the track here?

Reply:
Well the `setVaccinationDate` should be irrelevant - I think you can just compute it (e.g., `const createEnabled = vaccinationDate && ...`). So the only thing that matters are the current state(s) - not how (and in what order) they have been taken.

Reply:
Interesting! So if I get you right, you are saying  we can move this write this if clause in a different way out of `useEffect`

```
      const createEnabled = vaccinationDate && true; //want to be explicit here as true
      setEnableCreateButtons(createEnabled);
}
```
And it would reflect automatically whenever the vaccination date is selected.

Reply:
I have tried the above mentioned thing but it is throwing the error saying below as soon as I get it out of the `useEffect`:
> Uncaught Invariant Violation: Too many re-renders. React limits the number of renders to prevent an infinite loop.

Reply:
After some reading I thinlk I got what you meant here. You mean I should get rid of `enableCreateButtons/enableCreateButtons` etc as the states and use them more as local variables, which would be computed based on other states.
I have implemented it in the above way now.
Thanks